### PR TITLE
Adds ability to publish .jar to artifactory via env variables.

### PR DESCRIPTION
**Pull request by dougb**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -10,6 +10,28 @@ val circeVersion = "0.10.0"
 enablePlugins(GitVersioning)
 git.useGitDescribe := true
 
+val artifactoryUrl:java.net.URL = sys.env.get("ARTIFACTORY_URL").map(new java.net.URL(_)).getOrElse(new java.net.URL("http://localhost"))
```
Comment (colindean) [inline build.sbt]: Does this localhost thing actually work? Might it be better to have `artifactoryUrl` be an `Option[URL]` and the stuff below accesses it with `artifactoryUrl.get` or is only defined when the option isDefined?

Comment (dougb) [reply build.sbt]: No, it doesn't work. I made it an `Option[URL]` 
One of the problem is that I haven't figured out how to stand up an artifactory instance via Docker to test this pr.

**Comment by phpisciuneri** [comment]
@dougb lets get #13 merged.  Then we can do a release and give this a go, then we merge this one.

**Comment by dougb** [comment]
@phpisciuneri I want to merge this pr before we do a release. This will enable us to do publish it internally with a clean release tag `0.8.0` vs something like `0.8.0-1-g1743579`

**Pull request merged**